pkg/discovery/util/owner_util.go

```diff
@@ -19,28 +19,46 @@ type OwnerInfo struct {
 func GetOwnerInfo(owners []metav1.OwnerReference) (OwnerInfo, bool) {
 	var ownerInfo OwnerInfo
 	var ownerSet bool
-	for _, owner := range owners {
-		if owner.Controller == nil || !(*owner.Controller) {
-			glog.V(3).Infof("Owner %+v is not a managing controller.", owner)
-			continue
+	var controllerIndex int = -1
+	for i, owner := range owners {
+		if IsController(owner) {
+			controllerIndex = i
 		}
+
 		if len(owner.Kind) > 0 && len(owner.Name) > 0 && len(owner.UID) > 0 {
-			// Found a valid controller owner
+			if controllerIndex == i {
```

Comment:
Why can't we move the IsController(owner) check within the if len(owner.Kind) > 0 && len(owner.Name) > 0 && len(owner.UID) > 0 , so that we don't need the controllerIndex at all.

Reply:
Updated!